sycl/source/detail/event_impl.cpp

```diff
@@ -120,7 +121,7 @@ event_impl::event_impl(RT::PiEvent Event, const context &SyclContext)
   getPlugin().call<PiApiKind::piEventRetain>(MEvent);
 }
 
-event_impl::event_impl(QueueImplPtr Queue) {
+event_impl::event_impl(QueueImplPtr Queue) : MQueue{Queue} {
```

Comment:
Not sure what the point of that change would be. MQueue is a weak_ptr, so we would have to immediately create a shared_ptr in the function body.

Reply:
Is it feasible to create a weak_ptr from const reference to shared_ptr? If so, changing to const ref instead of instance of shared_ptr will save some resources with eliminating increment of atomic ref counter for shared_ptr.

Reply:
@s-kanaev Done.